packages/redux-query-react/src/hooks/use-memoized-query-config.js

```diff
@@ -26,13 +26,15 @@ const useMemoizedQueryConfig = (
 
     if (
       queryKey !== previousQueryKey.current ||
-      headersChanged([queryConfig], [previousQueryConfig.current])
+      (queryConfig != null &&
+        previousQueryConfig.current != null &&
+        headersChanged([queryConfig], [previousQueryConfig.current]))
     ) {
       previousQueryKey.current = queryKey;
       previousQueryConfig.current = queryConfig;
```

Comment:
```suggestion
      previousQueryConfig.current = providedQueryConfig;
```

Reply:
will change, not going to accept because keeping a single commit is better for rebasing in the future.

Reply:
oh, I see what you're saying I think